### PR TITLE
Release 1.89.0

### PR DESCRIPTION
**Pull request by TheanLim**
### Summary
* Enhancement - Bump google.golang.org/grpc from 1.64.0 to 1.64.1 in /ecs-agent/daemonimages/csidrive [#4361](https://github.com/aws/amazon-ecs-agent/pull/4361)
* Enhancement - Log unresolved dependencies at info level [#4419](https://github.com/aws/amazon-ecs-agent/pull/4419)
* Enhancement - Make the path to resolv.conf a configuration parameter instead of a hardcoded string [#4421](https://github.com/aws/amazon-ecs-agent/pull/4421)
* Bugfix - Implement reference counting of volume mounts in amazon-ecs-volume-plugin [#4425](https://github.com/aws/amazon-ecs-agent/pull/4425) 
* Bugfix - Set firelens mem_buf_limit by default [#4405](https://github.com/aws/amazon-ecs-agent/pull/4405)

### Testing

New tests cover the changes: 

### Additional Information

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  

**Does this PR include the addition of new environment variables in the README?**

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+# 1.89.0
+* Enhancement - Bump google.golang.org/grpc from 1.64.0 to 1.64.1 in /ecs-agent/daemonimages/csidrive [#4361](https://github.com/aws/amazon-ecs-agent/pull/4361)
```
Comment (mye956) [inline CHANGELOG.md:4]: super nit (non-blocking): typo in `csidrive` (missing `r`)

**Review by mye956 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (TheanLim) [reply CHANGELOG.md]: Thanks for catching that.

**Comment by prateekchaudhry** [comment]
Could you also update the version in this file - https://github.com/TheanLim/amazon-ecs-agent/blob/v1.89.0-stage/ecs-init/ECSVERSION

**Comment by TheanLim** [comment]
> Could you also update the version in this file - https://github.com/TheanLim/amazon-ecs-agent/blob/v1.89.0-stage/ecs-init/ECSVERSION
> 
> Or as a follow-up to #4423 we may actually remove ECSVERSION file

File removed

File: agent/version/version.go [hunk 2]
```diff
@@ -22,10 +22,10 @@ package version
...
 // Version is the version of agent
-const Version = "1.88.0"
+const Version = "1.89.0"
 
 // GitDirty indicates the cleanliness of the git repo when this agent was built
 const GitDirty = true
 
 // GitShortHash is the short hash of this agent build
-const GitShortHash = "c9040caf"
+const GitShortHash = "9494fb35"
```
Comment (danehlim) [inline agent/version/version.go:31]: This seems to point to commit `9494fb35`, however the latest commit in dev branch is `4843cd00` which I believe is the one that we want. 

I suspect this happened due to release PR being generated under a stale local branch which has not yet pulled the `4843cd00` commit changes from upstream. If I'm not mistaken, I think you just need to pull in the latest changes from dev into your local branch and re-perform the internal process to regenerate the `version.go` file(s).

File: ecs-init/version/version.go [hunk 3]
```diff
@@ -22,10 +22,10 @@ package version
...
 // Version is the version of ecs-init
-const Version = "1.88.0"
+const Version = "1.89.0"
 
 // GitDirty indicates the cleanliness of the git repo when this ecs-init was built
 const GitDirty = true
 
 // GitShortHash is the short hash of this ecs-init build
-const GitShortHash = "c9040caf"
+const GitShortHash = "9494fb35"
```
Comment (danehlim) [inline ecs-init/version/version.go:31]: Same comment here as: https://github.com/aws/amazon-ecs-agent/pull/4429#discussion_r1844383390

File: agent/version/version.go (hunk 2 above)
Comment (TheanLim) [reply agent/version/version.go:31]: Got it. Taking a look

Comment (TheanLim) [reply agent/version/version.go]: The branch wasn't stale but I forgot to rerun the internal process and regenerate the version files. Version files are regenerated now, pointing to `4843cd00`

**Review by prateekchaudhry — APPROVED** [review]

**Review by danehlim — APPROVED** [review]

**Pull request merged**